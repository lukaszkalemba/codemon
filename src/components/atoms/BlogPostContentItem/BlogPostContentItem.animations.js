const getContentItem = animate => {
  return {
    variants: {
      animate: {
        opacity: 1,
        y: -15,
        transition: { duration: 0.8 },
      },
      initial: { opacity: 0, y: 100 },
    },
    animate,
    initial: 'initial',
  };
};

export default { getContentItem };
