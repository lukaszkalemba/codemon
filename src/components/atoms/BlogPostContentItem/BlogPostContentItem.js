import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { useInView } from 'react-intersection-observer';
import { useAnimation } from 'framer-motion';
import S from './BlogPostContentItem.styles';
import animations from './BlogPostContentItem.animations';

const BlogPostContentItem = ({ content }) => {
  const animation = useAnimation();

  const [wrapperRef, inView] = useInView({
    threshold: 0.5,
  });

  useEffect(() => {
    if (inView) {
      animation.start('animate');
    }
  }, [animation, inView]);

  const wrapperAnimations = animations.getContentItem(animation);

  return (
    <S.Wrapper
      ref={wrapperRef}
      data-testid="blog-post-content-item"
      {...wrapperAnimations}
    >
      {content}
    </S.Wrapper>
  );
};

BlogPostContentItem.propTypes = {
  content: PropTypes.shape({}).isRequired,
};

export default BlogPostContentItem;
