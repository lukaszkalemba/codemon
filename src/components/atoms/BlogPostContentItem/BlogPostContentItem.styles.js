import styled from 'styled-components';
import { motion } from 'framer-motion';

const Wrapper = styled(motion.div)`
  h2 {
    font-size: ${({ theme }) => theme.font.size.heading.xxs};
    margin-top: 50px;
    margin-bottom: 5px;

    ${({ theme }) => theme.media.tablet`
      font-size: ${theme.font.size.heading.m};
      margin-top: 70px;
      margin-bottom: 10px;
    `}
  }

  p {
    font-size: ${({ theme }) => theme.font.size.body.s};
    max-width: 750px;
    line-height: 25px;
    opacity: 0.8;
    margin-bottom: 30px;

    ${({ theme }) => theme.media.desktop`
      max-width: 1000px;
    `}
  }
`;

export default { Wrapper };
