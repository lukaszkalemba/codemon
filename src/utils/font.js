export default {
  size: {
    xxxs: '1rem',
    xxs: '1.3rem',
    xs: '1.6rem',
    s: '2.2rem',
    m: '3rem',
    l: '4rem',
    xl: '5.4rem',
    xxl: '8.5rem',
    xxxl: '10rem',
  },

  weight: {
    regular: '400',
    medium: '500',
    bold: '700',
  },

  family: {
    montserrat: 'Montserrat',
  },
};
