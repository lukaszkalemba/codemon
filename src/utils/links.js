export default [
  {
    id: 0,
    path: '/projekty',
    name: 'projekty',
  },
  {
    id: 1,
    path: '/blog',
    name: 'blog',
  },
  {
    id: 2,
    path: '/wycena',
    name: 'wycena',
  },
];
