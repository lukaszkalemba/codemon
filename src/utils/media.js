export default {
  phone: '576px',
  tablet: '768px',
  desktop: '1150px',
  largeDesktop: '1600px',
};
