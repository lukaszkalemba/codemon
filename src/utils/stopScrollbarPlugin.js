import { ScrollbarPlugin } from 'smooth-scrollbar';

class stopScrollbarPlugin extends ScrollbarPlugin {
  static pluginName = 'stopScrollbar';

  static defaultOptions = {
    open: false,
  };

  transformDelta(delta) {
    return this.options.open ? { x: 0, y: 0 } : delta;
  }
}

export default stopScrollbarPlugin;
