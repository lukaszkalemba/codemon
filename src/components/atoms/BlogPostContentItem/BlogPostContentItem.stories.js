import React from 'react';
import { text, withKnobs } from '@storybook/addon-knobs';
import BlogPostContentItem from './BlogPostContentItem';

export const blogPostContentItem = () => (
  <BlogPostContentItem
    content={
      <>
        <h2>
          {text('Blog Post Content Item Title', 'Lorem ipsum dolor sit amet.')}
        </h2>
        <p>
          {text(
            'Blog Post Content Item Description',
            'Lorem ipsum, dolor sit amet consectetur adipisicing elit. Odit doloribus illum, aspernatur nulla id nihil suscipit consequuntur dolores expedita similique eveniet ex ullam, est libero beatae quam, placeat accusamus et.'
          )}
        </p>
      </>
    }
  />
);

export default {
  component: BlogPostContentItem,
  decorators: [withKnobs],
  title: 'Atoms|BlogPostContentItem',
};
