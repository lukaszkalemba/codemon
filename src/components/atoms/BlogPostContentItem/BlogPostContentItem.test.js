import React from 'react';
import { render } from 'utils/test-utils';
import '__mocks__/intersectionObserver';
import BlogPostContentItem from './BlogPostContentItem';

describe('<BlogPostContentItem />', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BlogPostContentItem content={<p>Lorem ipsum dolor sit amet.</p>} />
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
