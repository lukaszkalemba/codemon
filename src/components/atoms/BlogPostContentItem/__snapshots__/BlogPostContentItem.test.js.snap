// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogPostContentItem /> renders correctly and matches snapshot 1`] = `
<div
  class="sc-AxjAm epiZDW"
  style="opacity: 0; transform: translateY(100px) translateZ(0);"
>
  <p>
    Lorem ipsum dolor sit amet.
  </p>
</div>
`;
